src/PerformanceTests/Common/Scenarios/BaseRunner.cs

```diff
@@ -354,18 +449,51 @@ static IEnumerable<int> IterateUntilFalse(Func<bool> condition)
 
     protected async Task DrainMessages()
     {
-        var startCount = Statistics.Instance.NumberOfMessages;
+        const int DrainPollInterval = 1500;
+        NServiceBus.Performance.StatisticsBehavior.Shortcut = true;
+        var startCount = NServiceBus.Performance.StatisticsBehavior.ShortcutCount;
         long current;
+        var start = Stopwatch.StartNew();
 
         Log.Info("Draining queue...");
         do
         {
-            current = Statistics.Instance.NumberOfMessages;
-            Log.Debug("Delaying to detect receive activity...");
-            await Task.Delay(1000).ConfigureAwait(false);
-        } while (Statistics.Instance.NumberOfMessages > current);
+            current = NServiceBus.Performance.StatisticsBehavior.ShortcutCount;
+            Log.DebugFormat("Delaying to detect receive activity, last count is {0}...", current);
+            await Task.Delay(DrainPollInterval).ConfigureAwait(false);
+        } while (NServiceBus.Performance.StatisticsBehavior.ShortcutCount > current);
 
         var diff = current - startCount;
-        Log.InfoFormat("Drained {0} message(s)", diff);
+        Log.InfoFormat("Drained {0:N0} message(s) in {0:N0}ms", diff, start.ElapsedMilliseconds);
+        NServiceBus.Performance.StatisticsBehavior.Shortcut = false;
+    }
+
+    void ReadPermutationSeedDurationFactor()
```

Comment:
These two methods can probably be refactored into their own class?
These methods also aren't clear to me after 3 seconds of reading. Perhaps make them a bit more self explanatory?